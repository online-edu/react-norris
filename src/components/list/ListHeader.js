import React from 'react';
import PropTypes from 'prop-types';
import Switch from '../switch';
import Spinner from '../spinner';

/**
 * List header component.
 */
const ListHeader = ({
  title,
  loader,
  btnCaption: text,
  onBtnClick: click,
  switchCaption: svitch,
  onSwitchToggle,
}) => (
  <li className="list-group-item d-flex justify-content-between align-items-center jokes-list__item">
    <h5 className="mt-0 mb-1">{title}</h5>
    {loader && <Spinner />}
    {text && (
      <button type="button" className="btn btn-primary" onClick={click}>
        {text}
      </button>
    )}
    {svitch && (
      <Switch id={svitch} label={svitch} switchToggle={onSwitchToggle} />
    )}
  </li>
);
/**
 * List item default props.
 */
ListHeader.defaultProps = {
  loader: false,
  btnCaption: undefined,
  switchCaption: undefined,
  onSwitchToggle: () => {},
  onBtnClick: () => {},
};
/**
 * List item props types.
 */
ListHeader.propTypes = {
  /** Header title */
  title: PropTypes.string.isRequired,
  /** Button caption */
  btnCaption: PropTypes.string,
  /** Toggle spinner */
  loader: PropTypes.bool,
  /** Gets called when the button is clicked. */
  onBtnClick: PropTypes.func,
  /** Switch caption */
  switchCaption: PropTypes.string,
  /** Gets called when the switch is toggled. */
  onSwitchToggle: PropTypes.func,
};
export default ListHeader;
