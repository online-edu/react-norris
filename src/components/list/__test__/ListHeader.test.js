import React from 'react';
import { shallow, mount } from 'enzyme';
import ListHeader from '../ListHeader';

describe('<ListHeader />', () => {
  const onBtnClick = jest.fn();

  const wrapper = (
    <ListHeader title="Jokes" btnCaption="View More" onBtnClick={onBtnClick} />
  );

  it('renders correctly', () => {
    const component = shallow(wrapper);
    expect(component).toMatchSnapshot();
  });

  it('contains class name "list-group-item" ', () => {
    const component = shallow(wrapper);
    expect(component.find('.list-group-item')).toHaveLength(1);
  });

  it('contains a "button" ', () => {
    const component = shallow(wrapper);
    expect(component.find('button')).toHaveLength(1);
  });

  it('allows us to set props', () => {
    const component = mount(wrapper);
    expect(component.props().btnCaption).toEqual('View More');
    component.setProps({ btnCaption: 'Random' });
    expect(component.props().btnCaption).toEqual('Random');
  });

  it('simulates click events', () => {
    const component = shallow(wrapper);
    component.find('button').simulate('click');
    expect(onBtnClick.mock.calls.length).toBe(1);
  });
});
