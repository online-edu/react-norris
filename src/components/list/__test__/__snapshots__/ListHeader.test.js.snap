// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListHeader /> renders correctly 1`] = `ShallowWrapper {}`;
